apiserver/facades/agent/storageprovisioner/storageprovisioner.go

```diff
@@ -255,14 +258,25 @@ func NewStorageProvisionerAPIv4(
 // WatchApplications starts a StringsWatcher to watch CAAS applications
 // deployed to this model.
 func (s *StorageProvisionerAPIv4) WatchApplications(ctx context.Context) (params.StringsWatchResult, error) {
-	watch := s.st.WatchApplications()
+	watch, err := s.applicationService.WatchApplications(ctx)
+	if err != nil {
+		return params.StringsWatchResult{
+			Error: apiservererrors.ServerError(err),
+		}, err
+	}
 	if changes, ok := <-watch.Changes(); ok {
+		watcherID, err := s.watcherRegistry.Register(watch)
+		if err != nil {
+			return params.StringsWatchResult{
+				Error: apiservererrors.ServerError(err),
+			}, err
+		}
 		return params.StringsWatchResult{
-			StringsWatcherId: s.resources.Register(watch),
+			StringsWatcherId: watcherID,
 			Changes:          changes,
 		}, nil
 	}
-	return params.StringsWatchResult{}, watcher.EnsureErr(watch)
+	return params.StringsWatchResult{}, nil
```

Comment:
No error in this case?

Reply:
Changed this method to use the `EnsureRegisterWatcher` helper.